lib/azurex/blob.ex

```diff
@@ -44,10 +45,34 @@ defmodule Azurex.Blob do
       {:error, %HTTPoison.Response{}}
 
   """
-  @spec put_blob(String.t(), binary, String.t(), optional_string, keyword) ::
+  @spec put_blob(
+          String.t(),
+          binary() | {:stream, Enumerable.t()},
+          optional_string,
+          optional_string,
+          keyword
+        ) ::
           :ok
           | {:error, HTTPoison.AsyncResponse.t() | HTTPoison.Error.t() | HTTPoison.Response.t()}
-  def put_blob(name, blob, content_type, container \\ nil, params \\ []) do
+  def put_blob(name, blob, content_type, container \\ nil, params \\ [])
+
+  def put_blob(name, {:stream, bitstream}, _content_type, container, params) do
+    bitstream
+    |> Stream.transform(
+      fn -> [] end,
+      fn chunk, acc ->
+        with {:ok, block_id} <- Block.put_block(container, chunk, name, params) do
+          {[], [block_id | acc]}
+        end
+      end,
+      fn acc ->
+        Block.commit_list(acc, container, name, params)
+      end
+    )
+    |> Stream.run()
+  end
+
+  def put_blob(name, blob, content_type, container, params) do
```

Comment:
Does this for sure work with a nil content type?

Reply:
Good catch, uploading a blob as an entire binary didn't previously work with `nil` for content type. I've gone back and added support for it so that the blob will default to the `application/octet-stream` content type.